src/Base.js

```diff
@@ -0,0 +1,39 @@
+'use strict'
+
+class Base {
+  constructor (name, code, implementation, alphabet) {
```

Comment:
- [ ] @theobat thank you for the throughout explanation, can you make that be part of code comments and/or readme so that it is clear to everyone?